packages/docsify-server-renderer/src/utils.test.js

```diff
@@ -0,0 +1,24 @@
+/* eslint-disable no-global-assign */
+require = require('esm')(module /* , options */);
+const { expect } = require('chai');
+const { initJSDOM } = require('../../../test/_helper');
+const { isExternal } = require('./utils');
+
+describe('isExternal', () => {
+  it('detects external whether links are external', async () => {
+    initJSDOM('', {
+      url: 'http://127.0.0.1:3000',
+      runScripts: 'dangerously',
+      resources: 'usable',
+    });
+
+    expect(isExternal).to.be.instanceOf(Function);
+    expect(isExternal('/foo.md')).to.be.false;
+    expect(isExternal('//foo.md')).to.be.true;
+    expect(isExternal('//127.0.0.1:3000/foo.md')).to.be.false;
+    expect(isExternal('http://127.0.0.1:3001/foo.md')).to.be.true;
+    expect(isExternal('https://google.com/foo.md')).to.be.true;
+    expect(isExternal('//google.com/foo.md')).to.be.true;
+    expect(isExternal('/google.com/foo.md')).to.be.false;
+  });
+});
```

Comment:
I didn't get to circle back to this yet, but it is high on my priority list. These tests are the important part, so we know it is working well.